src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/WcfService.cs

```diff
@@ -234,5 +236,46 @@ public XmlCompositeType GetDataUsingXmlSerializer(XmlCompositeType composite)
             composite.StringValue = composite.StringValue;
             return composite;
         }
+
+        public Stream GetStream(string data)
+        {
+            return StringToStream(data);
+        }
+
+        public bool UploadStream(Stream stream)
+        {
+            string data = StreamToString(stream);
+
+            if (!string.IsNullOrWhiteSpace(data))
```

Comment:
See other comment -- I recommend we just return 'data' directly and let the test deal with testing whether it was right.